### PR TITLE
Add Nesting Navigators with Optional Auth

### PR DESCRIPTION
**Pull request by LucasBruzzone**
This PR adds the last type of navigation: Stack with Tab navigators and an optional Auth.

### CODE REVIEW

File: src/hooks/useSession.js [hunk 1]
```diff
@@ -1,10 +1,36 @@
...
+
+import { setAuthedScreen } from '@actions/authActions';
 
 const useSession = () => {
+  const route = useRoute();
+  const dispatch = useDispatch();
+  const navigation = useNavigation();
   const { token } = useSelector(({ authReducer }) => authReducer.toJS());
 
+
```
Comment (neocoaster) [inline src/hooks/useSession.js:14]:
```suggestion
```

File: src/screens/Login/index.js [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
-import { useDispatch } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
 import { useNavigation } from '@react-navigation/native';
 
 import Button from '@components/Button';
 import Input from '@components/Input';
 import useSetNavigationOptions from '@hooks/useSetNavigationOptions';
 
-import { SIGN_UP_SCREEN } from '@constants/screens';
+import { SIGN_UP_SCREEN, WELCOME_SCREEN, TAB_NAVIGATOR } from '@constants/screens';
```
Comment (neocoaster) [inline src/screens/Login/index.js:14]:
```suggestion
import {
  SIGN_UP_SCREEN,
  WELCOME_SCREEN,
  TAB_NAVIGATOR,
} from '@constants/screens';
```

File: src/screens/Login/index.js [hunk 3]
```diff
@@ -29,6 +29,29 @@ const Login = () => {
...
+        });
+      } else {
+        navigation.replace(TAB_NAVIGATOR, {
+          screen: WELCOME_SCREEN,
+        });
+      }
+    }
+  }, [token]);
+
+
```
Comment (neocoaster) [inline src/screens/Login/index.js:54]:
```suggestion
```

**Review by neocoaster — APPROVED** [review]

File: src/screens/Login/index.js (hunk 2 above)
Comment (LucasBruzzone) [reply src/screens/Login/index.js:14]: Due to this comment I added a restriction of max lines to the eslint so it is consistent throughout the code. This line does not exceed the maximum length

**Pull request merged**